File: AdamClassLib/Sample Code/APAX/WinCE/VB/APAX-PAC-Sample/APAX-5028/FormSafetySetting.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Public Class FormSafetySetting
    Public Delegate Sub EventHandler_ApplySafetyValueClick(ByVal szVal() As String)
    Public Event ApplySafetyValueClick As EventHandler_ApplySafetyValueClick
    Private m_iChannelTotal As Integer
    Private m_fVal() As Single
    Private m_szRange() As String

    Public Sub New()
        MyBase.New()
        InitializeComponent()
    End Sub

    Public Sub New(ByVal iChannelTotal As Integer, ByRef fVal() As Single, ByRef szRange() As String)
        MyBase.New()
        Dim lvItem As ListViewItem
        InitializeComponent()
        m_iChannelTotal = iChannelTotal
        m_fVal = fVal
        m_szRange = szRange
        ' Set init information
        Dim i As Integer = 0
        Do While (i < iChannelTotal)
            lvItem = New ListViewItem(i.ToString)
            lvItem.SubItems.Add(fVal(i).ToString)
            gridviewSafety.Items.Add(lvItem)
            i = (i + 1)
        Loop
    End Sub
   
    Private Sub gridviewSafety_SelectedIndexChanged_1(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles gridviewSafety.SelectedIndexChanged
        Dim i As Integer = 0
        Do While (i < gridviewSafety.SelectedIndices.Count)
            If gridviewSafety.Items(gridviewSafety.SelectedIndices(i)).Selected Then
                Dim iSelectedItem As Integer = gridviewSafety.SelectedIndices(i)
                txtChannel.Text = gridviewSafety.Items(iSelectedItem).Text
                txtSafetyVal.Text = gridviewSafety.Items(iSelectedItem).SubItems(1).Text
                labRange.Text = m_szRange(iSelectedItem)
                Return
            End If
            i = (i + 1)
        Loop
    End Sub

    Private Sub btnSet_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSet.Click
        Dim i As Integer = 0
        Do While (i < gridviewSafety.SelectedIndices.Count)
            If gridviewSafety.Items(gridviewSafety.SelectedIndices(i)).Selected Then
                Dim iSelectedItem As Integer = gridviewSafety.SelectedIndices(i)
                gridviewSafety.Items(iSelectedItem).SubItems(1).Text = txtSafetyVal.Text
                Return
            End If
            i = (i + 1)
        Loop
    End Sub

    Private Sub btnApply_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnApply.Click
        Try
            Dim o_szVal() As String = New String((m_iChannelTotal) - 1) {}
            Dim i As Integer = 0
            Do While (i < m_iChannelTotal)
                o_szVal(i) = Convert.ToString(gridviewSafety.Items(i).SubItems(1).Text)
                i = (i + 1)
            Loop
            RaiseEvent ApplySafetyValueClick(o_szVal)

        Catch ex As Exception
            MessageBox.Show("Setting Safety Fail", "Error")
        End Try

    End Sub
End Class

File: AdamClassLib/Sample Code/APAX/WinCE/CSharp/APAX-Coupler-Sample/APAX-5028/FormSafetySetting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace APAX_5028
{
    public partial class FormSafetySetting : Form
    {
        public delegate void EventHandler_ApplySafetyValueClick(string[] szVal);
        public event EventHandler_ApplySafetyValueClick ApplySafetyValueClick;
        int m_iChannelTotal;
        float[] m_fVal;
        string[] m_szRange;

        public FormSafetySetting(int iChannelTotal, float[] fVal, string[] szRange)
        {
            InitializeComponent();
            m_iChannelTotal = iChannelTotal;

            m_fVal = fVal;
            m_szRange = szRange;

            // Set init information
            for (int i = 0; i < iChannelTotal; i++)
            {
                ListViewItem lvItem = new ListViewItem(i.ToString());
                lvItem.SubItems.Add(fVal[i].ToString());
                gridviewSafety.Items.Add(lvItem);
            }
        }

        private void btnApply_Click(object sender, EventArgs e)
        {
            if (ApplySafetyValueClick != null)
            {
                string[] o_szVal = new string[m_iChannelTotal];
                for (int i = 0; i < m_iChannelTotal; i++)
                {
                    o_szVal[i] = Convert.ToString(gridviewSafety.Items[i].SubItems[1].Text);
                }
                ApplySafetyValueClick(o_szVal);
            }
        }

        private void btnSet_Click(object sender, EventArgs e)
        {
            for (int i = 0; i < gridviewSafety.SelectedIndices.Count; i++)
            {
                if (gridviewSafety.Items[gridviewSafety.SelectedIndices[i]].Selected)
                {
                    int iSelectedItem = gridviewSafety.SelectedIndices[i];
                    gridviewSafety.Items[iSelectedItem].SubItems[1].Text = txtSafetyVal.Text;
                    return;
                }
            }
        }

        private void gridviewSafety_SelectedIndexChanged(object sender, EventArgs e)
        {
            for (int i = 0; i < gridviewSafety.SelectedIndices.Count; i++)
            {
                if (gridviewSafety.Items[gridviewSafety.SelectedIndices[i]].Selected)
                {
                    int iSelectedItem = gridviewSafety.SelectedIndices[i];
                    txtChannel.Text = gridviewSafety.Items[iSelectedItem].Text;
                    txtSafetyVal.Text = gridviewSafety.Items[iSelectedItem].SubItems[1].Text;
                    labRange.Text = m_szRange[iSelectedItem];
                    return;
                }
            }
        }

    }
}